sphinx/screens-detail/create-tribe/create-tribe/src/main/java/chat/sphinx/create_tribe/ui/CreateTribeFragment.kt

```diff
@@ -241,9 +252,41 @@ internal class CreateTribeFragment: SideEffectFragment<
                     viewModel.imageLoaderDefaults
                 )
             }
-            CreateTribeViewState.CreatingTribe -> {
+            CreateTribeViewState.SavingTribe -> {
                 binding.progressBarCreateTribe.visible
             }
+            is CreateTribeViewState.LoadingExistingTribe -> {
+                binding.progressBarLoadTribeContainer.visible
+            }
+            is CreateTribeViewState.ExistingTribe -> {
+                binding.progressBarLoadTribeContainer.gone
+
+                binding.editTextTribeName.setText(viewState.name)
+
+                viewState.imageUrl?.let { imgUrl ->
+                    if (imgUrl.isNotEmpty()) {
+                        binding.editTextTribeImage.setText(imgUrl)
+                        imageLoader.load(
+                            binding.imageViewTribePicture,
+                            imgUrl,
+                            viewModel.imageLoaderDefaults
+                        )
+                    }
+                }
+
+                binding.editTextTribeTags.text = viewState.tags.joinToString { it }
+
+                binding.editTextTribeDescription.setText(viewState.description)
+
+                binding.editTextTribePriceToJoin.setText(viewState.priceToJoin.getStringOrEmpty())
```

Comment:
if calling `ViewModel.currentViewState`, you will not see the current text being presented to the user in the `editTextTribePriceToJoin` view, but will see a number value. The use of the extension function should be moved to where the `ViewState.priceToJoin` is set.

Definitely gotta get in the habit of building things out this way.